recipes/gstreamer/all/conanfile.py

```diff
@@ -24,187 +35,236 @@ class GStreamerConan(ConanFile):
         "fPIC": True,
         "with_introspection": False,
     }
-    generators = "pkg_config"
-    _meson = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
 
-    def requirements(self):
-        self.requires("glib/2.72.0")
+    def configure(self):
+        if self.options.shared or self.settings.os == "Windows":
```

Comment:
I have just "merged" the condtions from _configure_ and _config_options_ - but they are called independently or?

Reply:
They are called independently. Most notably to the user before or after passing the command line input.

Our usual template give the best UX.

- if you set fpic on windows it's a hard error (since that just isnt a thing)
- If you pass fpic on linux but also shared it just disgarded since the value need to be true

you need to pass one option instead of worrying about both + this matches the internal logic of the helpers

Reply:
Not entirely sure how its notably to the user, if its removed in either calls, but I have reverted to having both _configure_ and _config_options_. I'll try read about the two.

Reply:
Ahh, I think I get it now. If they are removed in config_options, you get an error if you try set it in your conan call, since its no longer an option.
If removed in configure, you won't get an error even if you tried to set it - hence it can be more difficult to actually figure out if the value to try to set is actually used or not.